OSDOCS-11158: adds image mode to MicroShift install (#80599)
openshift/openshift-docs: modules/microshift-install-rhel-image-mode-publish-image.adoc

// Module included in the following assemblies:
//
// microshift_install_bootc/microshift-install-rhel-image-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-rhel-image-mode-publish-image_{context}"]
= Publishing the bootc image to the remote registry

Publish your bootc image to the remote registry so that the image can be used for running the container on another host, or for when you want to install a new operating system with the bootc image layer.

.Prerequisites

> * A {op-system-base-full} 9.4 host with an active Red Hat subscription for building {microshift-short} bootc images and running containers.

Reviewer: Strictly speaking, the 1st requirement is not necessary for publishing images. 
You need to be logged into the same host where the local image was built.

Author: Carried over from building.

* You are logged into the {op-system} 9.4 host using the user credentials that have SUDO permissions configured.

> * You have a remote registry such as link:https:/quay.io[Red Hat quay] for storing and accessing bootc images.

Reviewer: ```suggestion
* You have a remote registry such as link:https://quay.io[Red Hat quay] for storing and accessing bootc images.
```

> * You created the `ContainerFile` and built the image.

Reviewer: ```suggestion
* You created the `Containerfile` and built the image.
```

.Procedure

. Log into your remote registry by running the following command:
+
[source,terminal]
----
REGISTRY_URL=quay.io

$ sudo podman login "${REGISTRY_URL}" # <1>
----
<1> Replace _REGISTRY_URL_ with the URL for your registry.

. Publish the image by running the following command:
+
[source,terminal]
----

> REGISTRY_IMG=myorg/mypath/"${IMAGE_NAME}" # <1> <2>

Reviewer: Should we also set the IMAGE_NAME here? It would work if the build and push is run in the same shell, but...

```suggestion
REGISTRY_IMG=myorg/mypath/"${IMAGE_NAME}" # <1> <2>
IMAGE_NAME=microshift-4.16-bootc
```

Author: Let's try it.

// Module included in the following assemblies:
//
// microshift_install_bootc/microshift-install-rhel-image-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-rhel-image-mode-publish-image_{context}"]
= Publishing the bootc image to the remote registry

Publish your bootc image to the remote registry so that the image can be used for running the container on another host, or for when you want to install a new operating system with the bootc image layer.

> .Prerequisites

Reviewer 2: ```suggestion
.Prerequisites

```

Reviewer 2: Should we have an empty line between Prerequisites and the list?

// Module included in the following assemblies:
//
// microshift_install_bootc/microshift-install-rhel-image-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-rhel-image-mode-publish-image_{context}"]
= Publishing the bootc image to the remote registry

Publish your bootc image to the remote registry so that the image can be used for running the container on another host, or for when you want to install a new operating system with the bootc image layer.

.Prerequisites

* You are logged into the {op-system} 9.4 host where the image was built using the user credentials that have `sudo` permissions.
* You have a remote registry such as link:https://quay.io[Red Hat quay] for storing and accessing bootc images.
* You created the Containerfile and built the image.

.Procedure

> . Log into your remote registry by running the following command:

Reviewer 3: ```suggestion
. Log in to your remote registry by running the following command:
```

+
[source,terminal]
----
REGISTRY_URL=quay.io

$ sudo podman login "${REGISTRY_URL}" # <1>
----
<1> Replace _REGISTRY_URL_ with the URL for your registry.

. Publish the image by running the following command:
+
[source,terminal]
----
REGISTRY_IMG=myorg/mypath/"${IMAGE_NAME}" # <1> <2>
IMAGE_NAME=microshift-4.16-bootc # <3>

$ sudo podman push localhost/"${IMAGE_NAME}" "${REGISTRY_URL}/${REGISTRY_IMG}"
----

> <1> Replace _<REGISTRY_IMG>_ with the name of your bootc image.

Reviewer 4: this seems like an unneeded note. since note 2 and 3 cover what goes into the variable REGISTRY_IMG

<2> Replace _<myorg/mypath>_ with your remote registry organization name and path.
<3> Replace _IMAGE_NAME_ with the name of the image you want to publish.

.Verification

> . Run the container using the image you pushed to your registry.

Reviewer: We should probably add something like: "as described below"? Otherwise, it's not clear how I should run it at this step of the docs.

Author: Yes, that makes sense. I'll just refer to the section title in case the content moves around at some point in the future.
